WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java

```diff
@@ -436,6 +437,19 @@ public boolean onPreferenceClick(Preference preference) {
         } else if (preference == mDeleteSitePref) {
             AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.SITE_SETTINGS_DELETE_SITE_ACCESSED, mSite);
             requestPurchasesForDeletionCheck();
+        } else if (preference == mRemoveSitePref) {
+            AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
+            dialogBuilder.setTitle(getResources().getText(R.string.remove_account));
+            dialogBuilder.setMessage(getResources().getText(R.string.sure_to_remove_account));
+            dialogBuilder.setPositiveButton(getResources().getText(R.string.yes), new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int whichButton) {
+                    mDispatcher.dispatch(SiteActionBuilder.newRemoveSiteAction(mSite));
+                    mSite = null;
```

Comment:
The problem with nullifying here is that the dispatch action is async and we are just assuming it'll be a successful one. I'd prefer it if we nullify the site in `onSiteRemoved` or one of the methods called from there. As far as I can tell from my debug sessions, the `onSiteRemoved` was getting called before the site is fetched again. You might want to verify that with Stetho first.